generators/app/templates/ext-ts/main.ts

```diff
@@ -1,11 +1,115 @@
 import { LensMainExtension } from "@k8slens/extensions";
 
-export default class ExampleExtensionMain extends LensMainExtension {
-    onActivate() {
-        console.log('helloworld-sample activated');
+// for demos of Lens' api, you only need LensMainExtension for
+// a minimum 'main' extension
+import { Store, Util } from "@k8slens/extensions";
+
+/**
+ * LensMainExtension is the interface to Lens' main process
+ * 
+ * LensMainExtension api allows you to access, configure, and customize Lens data add
+ * custom application menu items, and generally run custom code in Lens'
+ * main process.
+ * 
+ * See more details: <https://docs.k8slens.dev/master/extensions/capabilities/common-capabilities/#main-extension>
+ * LensMainExtension API doc <https://docs.k8slens.dev/master/extensions/api/classes/lensmainextension/>
+ * 
+ * MainExtension which extends LensMainExtension runs in Lens' 'main' process (NOT 'renderer' process)
+ * main vs renderer <https://www.electronjs.org/docs/tutorial/quick-start#main-and-renderer-processes>
+ * !! Note that the console statements in MainExtension is NOT visible in the 
+ * !! DevTools console in Lens
+ * To see console statements, start the Lens app from a Terminal on Mac OSX:
+ * `/Applications/Lens.app/Contents/MacOS/Lens`
+ * 
+ * On Linux, you can access the Main process logs using the Lens PID. First get the PID
+ * `ps aux | grep Lens | grep -v grep`
+ * 
+ * And get the log
+ * `tail -f /proc/[pid]/fd/1 # stdout (console.log)`
+ * `tail -f /proc/[pid]/fd/2 # stderr (console.error)`
+ * 
+ * TODO: Add instruction on how-to to see main process logs on Windows
+ * @export
+ * @class MainExtension
+ * @extends {LensMainExtension}
+ */
+export default class MainExtension extends LensMainExtension {
+    // Private instance fields used in onActivate/onDeactivate
+    #timer: NodeJS.Timeout
+    
+    /**
+     * onActivate is called when your extension has been successfully enabled.
+     * You can add your custom logic here, code inside onActivate block will be called 
+     * when your extension is enabled. (typically from the Lens Extensions Page)
+     * @memberof MainExtension
+     */
+    onActivate(): void {
+      // print hello world when extension is activated
+      // !! Note that the console statements in MainExtension is NOT visible in the 
+      // !! DevTools console in Lens
+      // To see console statements, start the Lens app from a Terminal
+      console.log("activated");
+
+      // The following block is an example you can do in onActivate()
+      // You can access Lens' state data and periodically logs the name of the 
+      // currently active cluster in Lens.
+      // 
+      // see what else you can do with Store in our API docs
+      // <https://docs.k8slens.dev/master/extensions/api/modules/_core_api_stores_/>
+      const { clusterStore } = Store;
+      this.#timer = setInterval(() => {
```

Comment:
Interval could be replaced with `reaction` but maybe `setInterval` is easier to understand 🤔

Reply:
I think setinterval is easier for average js developer like me

Reply:
I agree, this is in the main extension guide, keeps it simple to show you can setup/teardown and gives a first, basic exposure to Stores, to show something useful you can do.